code/modules/events/disease_outbreak.dm

```diff
@@ -1,41 +1,75 @@
+	var/static/list/diseases_minor = list()
+	var/static/list/diseases_moderate = list()
+	var/static/list/diseases_major = list()
 
 /datum/event/disease_outbreak/setup()
+	if(isemptylist(diseases_minor) && isemptylist(diseases_moderate) && isemptylist(diseases_major))
+		populate_diseases()
+	if(isemptylist(transmissable_symptoms))
+		populate_symptoms()
+	var/datum/disease/virus
 	if(prob(25))
-		var/virus_type = pick(/datum/disease/advance/flu, /datum/disease/advance/cold, /datum/disease/brainrot, /datum/disease/magnitis, /datum/disease/beesease, /datum/disease/anxiety, /datum/disease/fake_gbs, /datum/disease/fluspanish, /datum/disease/pierrot_throat, /datum/disease/lycan)
-		D = new virus_type()
+		switch(severity)
+			if(EVENT_LEVEL_MUNDANE) virus = pick(diseases_minor)
+			if(EVENT_LEVEL_MODERATE) virus = pick(diseases_moderate)
+			if(EVENT_LEVEL_MAJOR) virus = pick(diseases_major)
+		D = new virus()
 	else
-		var/datum/disease/advance/A = new /datum/disease/advance
-		A.name = capitalize(pick(GLOB.adjectives)) + " " + capitalize(pick(GLOB.nouns + GLOB.verbs)) // random silly name
-		A.GenerateSymptoms(1,9,6)
-		A.AssignProperties(list("resistance" = rand(0,11), "stealth" = rand(0,2), "stage_rate" = rand(0,5), "transmittable" = rand(0,5), "severity" = rand(0,10)))
-		D = A
-
+		D = create_virus(severity * 2)	//Severity of the virus depends on the event severity level
 	D.carrier = TRUE
 
-/datum/event/disease_outbreak/announce()
-	GLOB.event_announcement.Announce("Confirmed outbreak of level 7 major viral biohazard aboard [station_name()]. All personnel must contain the outbreak.", "Biohazard Alert", 'sound/AI/outbreak7.ogg')
-
 /datum/event/disease_outbreak/start()
-	for(var/mob/living/carbon/human/H in shuffle(GLOB.alive_mob_list))
-		if(!H.client)
-			continue
-		if(issmall(H)) //don't infect monkies; that's a waste
-			continue
-		var/turf/T = get_turf(H)
-		if(!T)
-			continue
-		if(!is_station_level(T.z))
-			continue
+	GLOB.current_pending_disease = D
+	var/severity_text = "undefined (contact a coder)"
+	switch(severity)
+		if(EVENT_LEVEL_MUNDANE) severity_text = "mundane"
+		if(EVENT_LEVEL_MODERATE) severity_text = "moderate"
+		if(EVENT_LEVEL_MAJOR) severity_text = "major"
+
+
+	for(var/mob/M as anything in GLOB.dead_mob_list) //Announce outbreak to dchat
+		to_chat(M, "<span class='deadsay'><b>Disease outbreak:</b> The next new arrival is a carrier of a [severity_text] disease!</span>")
 
-		if(!H.ForceContractDisease(D))
+//Creates a virus with a harmful effect, guaranteed to be spreadable by contact or airborne
+/datum/event/disease_outbreak/proc/create_virus(max_severity = 6)
+	var/datum/disease/advance/A = new /datum/disease/advance
+	A.symptoms = A.GenerateSymptomsBySeverity(max_severity - 1, max_severity, 2) //Choose "Payload" symptoms
+	A.AssignProperties(A.GenerateProperties())
+	var/list/symptoms_to_try = deepCopyList(transmissable_symptoms)
+	for(var/sanity=0, sanity<25, sanity++)
```

Comment:
Unless I'm blind, it doesn't look like you use the iteration variable anywhere, and while I get that might be an expression of the coder, it's a *bit* confusing?
```suggestion
	for(var/i in 1 to 25)
```

Reply:
Well, I do wanted to express that the variable is there as a protection against infinite loops (sanity check).
If I make it "i" 25 becomes a bit of a magic number with no explanation.

I can make a define to document the intention, or put the explanation in the comment.